Automattic/wp-calypso
# Post type list: Multiple Selection Mode

This PR adds the ability to select multiple posts from a `PostTypeList` for the `/posts` section.

It is behind two feature-flags and part of a larger epic. See p8F9tW-fx-p2 and p8F9tW-vh-p2.

Testing instructions:
* `ENABLE_FEATURES=posts/post-type-list,posts/post-type-list/bulk-edit npm start`
* Visit the "Blog Posts" page using the sidebar in Calypso.
* Next to the search icon in the section nav, click on "Select" to toggle on selection mode. 
![image](https://user-images.githubusercontent.com/363749/31620994-4a62fb30-b25e-11e7-82d2-9a2d018a4e94.png)

* You should see a new bulk edit bar appear beneath the section nav.
![image](https://user-images.githubusercontent.com/363749/31621017-57c4d3a2-b25e-11e7-8294-787eed2a549a.png)

* You should see checkboxes appear alongside the post items in the list.
![image](https://user-images.githubusercontent.com/363749/31621039-63589a64-b25e-11e7-96fa-04ccd490a5a3.png)

* Checking a checkbox should increment the count in the bulk edit bar.
![image](https://user-images.githubusercontent.com/363749/31621053-6c5c9ec6-b25e-11e7-8758-f03d4f9ae5f0.png)

* Unchecking a checkbox should decrement the count.

![image](https://user-images.githubusercontent.com/363749/31621066-74d83a6a-b25e-11e7-94ee-667dfea3d720.png)

* Clicking "Select" again should toggle the selection mode off, hiding the bulk edit bar and the checkboxes.
* Exiting bulk edit mode and re-entering should clear the previous selection.

Notes: The edit and delete buttons do not function. This will be handled by a future story.
